src/networking.c

```diff
@@ -3438,6 +3438,28 @@ sds getAllClientsInfoString(int type, int hide_user_data) {
     return o;
 }
 
+sds getAllFilteredClientsInfoString(clientFilter *client_filter, int hide_user_data) {
+    listNode *ln;
+    listIter li;
+    client *client;
+    sds o = sdsnewlen(SDS_NOINIT, 200 * listLength(server.clients));
+    sdsclear(o);
+    listRewind(server.clients, &li);
+    while ((ln = listNext(&li)) != NULL) {
+        client = listNodeValue(ln);
+        if (client_filter->addr && strcmp(getClientPeerId(client), client_filter->addr) != 0) continue;
+        if (client_filter->laddr && strcmp(getClientSockname(client), client_filter->laddr) != 0) continue;
+        if (client_filter->type != -1 && getClientType(client) != client_filter->type) continue;
+        if (client_filter->id != 0 && client->id != client_filter->id) continue;
+        if (client_filter->user && client->user != client_filter->user) continue;
+        if (client_filter->skipme) continue;
+        if (client_filter->max_age != 0 && (long long)(commandTimeSnapshot() / 1000 - client->ctime) < client_filter->max_age) continue;
```

Comment:
We should have a function just for checking if a client matches a filter and returns 1 or 0. Then it can be used in CLIENT LIST and CLIENT KILL.

```C
int isClientMatchingFilter(client *c, clientFilter *client_filter);
```

Reply:
yeah, I was considering the same. The difference is KILL is not allowed to close the origin client connection, but can take that condition out separately.

Reply:
Exactly, it is better that the function just checks for a match, instead of creating info or killing the user, so it can be used by both commands.

Reply:
> The difference is KILL is not allowed to close the origin client connection, but can take that condition out separately.

I'm not sure I understand what you mean, but I added a comment about the CLIENT KILL code just to be clear.

Reply:
Yes, we are on the same page. Thank you!